src/datadog/extraction_util.cpp

```diff
@@ -287,14 +288,32 @@ ExtractedData merge(const std::vector<ExtractedData>& contexts) {
                data.trace_id == found->trace_id;
       });
 
+  const auto dd = std::find_if(
+      contexts.begin(), contexts.end(), [&](const ExtractedData& data) {
+        return data.style == PropagationStyle::DATADOG &&
+               data.trace_id == found->trace_id;
+      });
+
   if (other != contexts.end()) {
-    result.datadog_w3c_parent_id = other->datadog_w3c_parent_id;
     result.additional_w3c_tracestate = other->additional_w3c_tracestate;
     result.additional_datadog_w3c_tracestate =
         other->additional_datadog_w3c_tracestate;
     result.headers_examined.insert(result.headers_examined.end(),
                                    other->headers_examined.begin(),
                                    other->headers_examined.end());
+
+    if (result.parent_id != other->parent_id) {
+      if (other->datadog_w3c_parent_id &&
+          other->datadog_w3c_parent_id != "0000000000000000") {
```

Comment:
That condition was not needed before, why is it now?

Reply:
This condition that checks for value `"0000000000000000"` is needed now because if the `tracestate[dd][p]` value is not present or that default value, we must default to the Datadog `x-datadog-parent-id` value

Reply:
Resolving, but feel free to re-open if you have any further questions